libs/governance/src/lib/proposals-list/use-column-defs.tsx

```diff
@@ -0,0 +1,167 @@
+import { useMemo } from 'react';
+import BigNumber from 'bignumber.js';
+import type { ColDef } from 'ag-grid-community';
+import { useEnvironment } from '@vegaprotocol/environment';
+import {
+  t,
+  NetworkParams,
+  useNetworkParams,
+  getDateTimeFormat,
+} from '@vegaprotocol/react-helpers';
+import type {
+  VegaICellRendererParams,
+  VegaValueFormatterParams,
+} from '@vegaprotocol/ui-toolkit';
+import { ExternalLink } from '@vegaprotocol/ui-toolkit';
+import { ProposalStateMapping } from '@vegaprotocol/types';
+import type { ProposalListFieldsFragment } from '../proposals-data-provider/__generated___/Proposals';
+import { VoteProgress } from '../voting-progress';
+
+type NewMarket = {
```

Comment:
this type can be generated from fragment

```
fragment NewMarketFields on NewMarket {
  instrument {
    code
    name
    futureProduct {
      settlementAsset {
        id
        name
        symbol
      }
    }
  }
}

fragment ProposalListFields on Proposal {
  id
  reference
  state
  datetime
  votes {
    yes {
      totalTokens
      totalNumber
      totalWeight
    }
    no {
      totalTokens
      totalNumber
      totalWeight
    }
  }
  terms {
    closingDatetime
    enactmentDatetime
    change {
      ... on NewMarket {
        ...NewMarketFields
      }
    }
  }
}

query ProposalsList($proposalType: ProposalType, $inState: ProposalState) {
  proposalsConnection(proposalType: $proposalType, inState: $inState) {
    edges {
      node {
        ...ProposalListFields
      }
    }
  }
}
```

Reply:
yes, ofc, I knew it, thanks for reminding me